api/test/unit/controllers/test_search_controller.py

```diff
@@ -13,36 +13,51 @@
 
 
 @pytest.mark.parametrize(
-    "total_hits, real_result_count, page_size, expected",
+    "total_hits, real_result_count, page_size, page, expected",
     [
         # No results
-        (0, 0, 10, (0, 0)),
+        (0, 0, 10, 0, (0, 0)),
         # Setting page size to 0 raises an exception
         pytest.param(
-            0, 0, 0, (0, 0), marks=pytest.mark.raises(exception=ZeroDivisionError)
+            5, 5, 0, 0, (0, 0), marks=pytest.mark.raises(exception=ZeroDivisionError)
         ),
         # Fewer results than page size leads to max of result total
-        (5, 5, 10, (5, 0)),
-        # Even if no real results exist, total result count and page count are returned
-        # (seems like an impossible case IRL)
-        (100, 0, 10, (100, 10)),
+        (5, 5, 10, 0, (5, 1)),
+        # If no real results exist, even if there are hits, fall back to 0, 0
+        # (This case represents where all the links for a result are dead, for example)
+        (100, 0, 10, 0, (0, 0)),
         # If there are real results and ES reports no hits, nothing is expected
         # (seems like an impossible case IRL)
-        (0, 100, 10, (0, 0)),
+        (0, 100, 10, 0, (0, 0)),
         # Evenly divisible number of pages
-        (25, 5, 5, (25, 5)),
+        (25, 5, 5, 0, (25, 5)),
         # Unevenly divisible number of pages
-        (21, 5, 5, (21, 5)),
+        (21, 5, 5, 0, (21, 5)),
         # My assumption would be that this yields (20, 4), but the code is such that
         # when the "natural" page count can't be cleanly divisible by the page size,
         # We increment it plus one. Why would that be the case? 20 results, with 5
         # results per-page, would seem to result in 4 pages total not 5 🤷‍♀️
-        (20, 5, 5, (20, 5)),
+        (20, 5, 5, 0, (20, 4)),
```

Comment:
I guess we can remove the comment above now?

Reply:
Done in https://github.com/WordPress/openverse-api/pull/1032/commits/bfa0802d00cdc4282e2063b8f91fac8e66f1ff9a